### PR TITLE
Instructions on how to download samples

### LINKED ISSUE #1644: Need: Instructions on how to download samples
The problem: We tell customers to download the sample. We provide a link to the sample, but you can't download from the sample link. You need to figure out:

Navigate to repro.
Download the entire repro (and probably unzip)
navigate to the sample
ACOM had re-usable sections - How did ACOM migrate without shared/included sections. That's what we need here. Should we request this feature? Until it comes, we need one document explaining the process. That doc should probably live in the .NET repo.

Original issue: https://github.com/aspnet/Docs/issues/565

cc @dend

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #1644

### CODE REVIEW

**Comment by GuardRex** [comment]
@Rick-Anderson @mairaw @tdykstra @BillWagner See if I'm headed in the right direction ...

I make the *Samples and tutorials* doc repo agnostic by adding a section for ASP.NET Core tutorials. Also, its instructions for viewing/downloading samples is repo agnostic. In this form, it can be linked from both repos and saves having to create a separate doc just for viewing/downloading instructions.

Note that several of these, e.g. ...
* docs/csharp/expression-trees-building.md
* docs/csharp/indexers.md
* docs/csharp/tutorials/console-webapiclient.md

... are in 1st person. I didn't try to fix that on this pass.

Unless you want to go in a different direction, it's ready for review.

File: docs/samples-and-tutorials/index.md
Comment (tdykstra) [inline docs/samples-and-tutorials/index.md]: I hadn't seen this page in a while and am confused about the purpose of these sections as they stand.  From the topic intro and headings I would expect here a list of all or perhaps most tutorials/samples available. But there's just one .NET Core sample and two tutorials. 

* What are the selection criteria behind the .NET Core lists of samples and tutorials?
* Instead of linking to individual tutorials can we link to the respective tutorials pages? Or put all the tutorials links in an include that is referenced from both places? 
* Are the samples lists supposed to be complete?  If they're a subset also, perhaps we could explain what's excluded.
* I'd consider putting the how-to-download section at the end and then link to that anchor from sample links in docs.  It seems obtrusive to put it up front when it isn't really the primary focus of the article.

File: docs/samples-and-tutorials/index.md
Comment (tdykstra) [inline docs/samples-and-tutorials/index.md]: Samples are in other docs also.  You could mention that many of the ASP.NET Core articles that aren't tutorials also have samples associated with them

**Comment by tdykstra** [comment]
BTW Some of my suggestions would require significant amount of extra effort to fix pre-existing issues unrelated to providing download instructions -- it's fine to defer those to a separate issue.  This is a substantial effort already and the new instructions and link text changes look great.  Many thanks for taking on this task that we've needed done for a long time now!

File: docs/samples-and-tutorials/index.md
Comment (GuardRex) [inline docs/samples-and-tutorials/index.md]: > I'd consider putting the how-to-download section at the end and then link to that anchor from sample links in docs. It seems obtrusive to put it up front when it isn't really the primary focus of the article.

I originally had it at the end. I moved it up because you probably need to be aware of this information *before* you start clicking on the sample repos. Otherwise, there could (a) be a single sentence in the opening paragraph that says "for info on downloading, see section" and link it to the section at the bottom, or (b) every one of these sample README's need a blurb about how to get the code downloaded and that would be linked back to that section at the bottom of this doc.

**Comment by GuardRex** [comment]
@tdykstra np. I'm glad I'm at least moving in the right direction. I really don't mind taking the time to fish all of the tutorials/samples up and list them all here ... or as you say, scrap those parts and just link to existing tutorials pages, such as the ASP.NET Core one. [idk if there is a corollary yet for .NET Core tutorials ... I haven't looked yet]

I'll just fix the small wording changes you mentioned. I won't do anything else until I get my marching orders.

File: docs/samples-and-tutorials/index.md
Comment (tdykstra) [inline docs/samples-and-tutorials/index.md]: @Rick-Anderson @mairaw What do you think?  Seems to me the expectation set up by the title "Samples and Tutorials" is a list of samples and tutorials, and it would feel strange to be confronted with a long "how to download" section before I saw my first actual link to a sample.  A short note "Find [instructions for downloading](#) later in the article" doesn't seem problematic to me.

**Comment by GuardRex** [comment]
@tdykstra I made the small changes, including moving the section.

I'm leaning toward changing the section name from *Viewing or downloading samples* to *Viewing and downloading samples* because the instructions for both are in the section. If the section was about making a choice between the two, then I'd probably stick with "or." Do you agree it should be "and"?

**Comment by GuardRex** [comment]
@tdykstra Do you think there's enough good karma on this one to get it reviewed and merged? Is it the type of PR that should have multiple reviewers for sign-off on the enhancements?

File: docs/core/tutorials/netcore-hosting.md
Comment (rpetrusha) [inline docs/core/tutorials/netcore-hosting.md]: Nit: I've never liked "please see". I'd change this to "...build a Windows host; see the notes at the end..."

File: docs/core/tutorials/netcore-hosting.md
Comment (rpetrusha) [inline docs/core/tutorials/netcore-hosting.md]: in dotnet/docs --> in the dotnet/docs

File: docs/csharp/indexers.md
Comment (rpetrusha) [inline docs/csharp/indexers.md]: I don't see the value of providing the location for all samples, followed by the location for the samples related to this topic. I'd remove the sentence beginning "The code for all..." And rephrase the following sentence (remove "Otherwise", remove "directly".

File: docs/csharp/tutorials/console-teleprompter.md
Comment (rpetrusha) [inline docs/csharp/tutorials/console-teleprompter.md]: The first feature to add is the ability to...
Also, no comma between "file" and "and"

File: docs/csharp/tutorials/console-webapiclient.md
Comment (rpetrusha) [inline docs/csharp/tutorials/console-webapiclient.md]: redundancy in "final sample" and "sample app". Perhaps "with the final sample for this topic, you can download it."

File: docs/samples-and-tutorials/index.md
Comment (rpetrusha) [inline docs/samples-and-tutorials/index.md]: I'd tone down value-laden/subjective adjectives (rich, great).

File: docs/samples-and-tutorials/index.md
Comment (rpetrusha) [inline docs/samples-and-tutorials/index.md]: object-oriented (hyphen and lowercase)

File: docs/samples-and-tutorials/index.md
Comment (rpetrusha) [inline docs/samples-and-tutorials/index.md]: task (lowercase)

**Review by rpetrusha — CHANGES_REQUESTED** [review]

**Comment by GuardRex** [comment]
@rpetrusha Sorry for the commit noise. My GH local has taken a :poop: on me, so I'll see what's what with that later. I made your changes.

File: docs/samples-and-tutorials/index.md
Comment (tdykstra) [inline docs/samples-and-tutorials/index.md]: Or shorten to 

> To view a sample, just follow the sample link.

File: docs/samples-and-tutorials/index.md
Comment (tdykstra) [inline docs/samples-and-tutorials/index.md]: that contains

~~for your local system~~

File: docs/samples-and-tutorials/index.md
Comment (tdykstra) [inline docs/samples-and-tutorials/index.md]: to the ~~.NET~~ documentation
system that you won't --> system, you won't

File: docs/samples-and-tutorials/index.md
Comment (tdykstra) [inline docs/samples-and-tutorials/index.md]: Consider making this one first, It's simplest and probably most appropriate for most people.

File: docs/samples-and-tutorials/index.md
Comment (tdykstra) [inline docs/samples-and-tutorials/index.md]: Navigate within the repository to the sample's location. --> Navigate within the repository's files and folders to the sample's location.

appears ~~in the GitHub UI and~~ in your browser's address bar

File: docs/samples-and-tutorials/index.md
Comment (tdykstra) [inline docs/samples-and-tutorials/index.md]: Consider omitting.  This should be self-evident that once you're at a folder in the file system.

**Review by tdykstra — APPROVED** [review]
Looks great!  I added a few more comments and suggestions.

**Comment by tdykstra** [comment]
@GuardRex 

> Do you think there's enough good karma on this one to get it reviewed and merged? Is it the type of PR that should have multiple reviewers for sign-off on the enhancements?

I'd say it's OK to merge once Ron approves. We can always continue to tweak it later.

**Comment by GuardRex** [comment]
@tdykstra Sounds good. I'll get all of the feedback in there shortly.

**Comment by GuardRex** [comment]
@tdykstra @rpetrusha ... changes made.

**Comment by tdykstra** [comment]
LGTM

**Review by rpetrusha — APPROVED** [review]

**Pull request merged**
